Sources/Articles/Documentation.docc/2024/swiftpm-snippets/SwiftPM Snippets.md

```diff
@@ -0,0 +1,264 @@
+
+Next, create a new directory named `Snippets` at the top level of the package.
+
+```bash
+$ mkdir Snippets
+```
+
+>   Note: Despite the specification of [SE-0356](https://github.com/apple/swift-evolution/blob/main/proposals/0356-swift-snippets.md#overriding-the-location-of-snippets), it is not possible to name the directory anything other than `Snippets`.
+
+Inside the `Snippets` directory, create a new Swift file named `SnippetsExample_I.swift`.
+
+@Snippet(id: "SnippetsExample_I")
+
+You should now have a directory structure that looks like this:
+
+```
+📂 swift-snippets
+├── 📂 Snippets
+│   └── 📄 SnippetsExample_I.swift
+├── 📂 Sources
+│   └── 📂 SnippetsExample
+│       └── 📄 anchor.swift
+├── 📄 Package.swift
+└── 📄 .gitignore
+```
+
+That’s it!
+
+
+## Running Snippets
+
+The `swift build` command will automatically discover and compile all Snippets in a package.
+
+Snippets are just modules with discovery enabled, which means you can also build a single Snippet with the `--target` flag.
+
+```bash
+$ swift build --target SnippetsExample_I
+```
+
+You can run a Snippet with the `swift run` command.
+
+```bash
+$ swift run SnippetsExample_I
+```
+
+```text
+Building for debugging...
+[5/5] Linking SnippetsExample_I
+Build complete! (0.97s)
+Hi Barbie!
+```
+
+You could also run a Snippet with release optimizations, just like any other executable target.
+
+```bash
+$ swift run -c release SnippetsExample_I
+```
+
+```text
+Building for production...
+[17/17] Linking SnippetsExample_I
+Build complete! (9.73s)
+Hi Barbie!
+```
+
+>   Note:
+>   If the Snippet name contains special characters, you should pass the file name as-is to the `--target` flag. For example, to run a Snippet named `Foo bar.swift`, you would use `swift run 'Foo bar'` and not `Foo_bar`.
+
+
+## Embedding Snippets in documentation
+
+Most modern documentation engines support embedding Snippets in Markdown documentation via the `@Snippet` block directive.
+
+Let’s create a documentation bundle for the `SnippetsExample` target.
+
+```bash
+$ mkdir -p Sources/SnippetsExample/docs.docc
+```
+
+>   Tip:
+>   You can name the documentation bundle anything you like, as long as it has a `.docc` extension. The name comes from the DocC tool, but most other documentation engines will recognize it too.
+
+Create a markdown article named `My article.md` in the `docs.docc` directory.
+
+@Code(name: "My article.md", file: "My article (2).md.txt")
+
+In the example above, we have specified the Snippet to include by `path` identity. Despite its naming, the `path` syntax is not a file path. The first component is the name of the package **as specified by the ``PackageDescription/Package/name`` field in the manifest**. The second component is always the string `Snippets`. The third component is the Snippet ID, which is the name of the Snippet file without the `.swift` extension. If the Snippet ID contains special characters, you should pass the ID as-is, without replacing any characters.
```

Comment:
```suggestion
In the example above, the Snippet is included by using the `path` argument. 

Despite its naming, the `path` syntax is not a file path. The first component is the name of the package **as specified by the ``PackageDescription/Package/name`` field in the manifest**. The second component is always the string `Snippets`. The third component is the Snippet ID, which is the name of the Snippet file without the `.swift` extension.

If the Snippet ID contains special characters, the ID should be passed as-is, without replacing any characters.
```

Reply:
broke up the paragraph, but kept active voice